File: src/main/java/com/menmar/gestionTienda/service/EstadisticaService.java
package com.menmar.gestionTienda.service;

import com.menmar.gestionTienda.model.estadistica.ResumenEstadisticaResponse;
import com.menmar.gestionTienda.model.estadistica.TicketPendienteRecogidaResponse;
import com.menmar.gestionTienda.model.PageResponse;
import org.springframework.data.domain.Pageable;

import java.time.LocalDate;

public interface EstadisticaService {
    ResumenEstadisticaResponse resumen(LocalDate desde, LocalDate hasta, Long establecimientoId);
    PageResponse<TicketPendienteRecogidaResponse> pendientesRecogida(int diasDesde, Long establecimientoId, Pageable pageable);
}

File: frontend/src/service/EstadisticaService.ts
import api from './ApiService'
import type { PageResponse, ResumenEstadisticaResponse, TicketPendienteRecogidaResponse } from '@/types'

const EstadisticaService = {
    async resumen(
        desde: string,
        hasta: string,
        establecimientoId?: number
    ): Promise<ResumenEstadisticaResponse> {
        const res = await api.get<ResumenEstadisticaResponse>('/estadisticas/resumen', {
            params: { desde, hasta, establecimientoId }
        })
        return res.data
    },

    async pendientesRecogida(
        diasDesde?: number,
        establecimientoId?: number,
        page = 0,
        size = 20
    ): Promise<PageResponse<TicketPendienteRecogidaResponse>> {
        const res = await api.get<PageResponse<TicketPendienteRecogidaResponse>>(
            '/estadisticas/pendientes-recogida',
            { params: { diasDesde, establecimientoId, page, size } }
        )
        return res.data
    }
}

export default EstadisticaService
